src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java

```diff
@@ -1,11 +1,162 @@
+
+        turretMotor.configForwardSoftLimitThreshold(MAX_TURRET_ANGLE * DEGREES_TO_ENCODER_TICKS);
+        turretMotor.configReverseSoftLimitThreshold(MIN_TURRET_ANGLE * DEGREES_TO_ENCODER_TICKS);
+        turretMotor.configForwardSoftLimitEnable(true);
+        turretMotor.configReverseSoftLimitEnable(true);
+        turretMotor.configSupplyCurrentLimit(turretCurrentLimit);
+        turretMotor.setNeutralMode(NeutralMode.Brake);
+        turretMotor.configSelectedFeedbackSensor(TalonFXFeedbackDevice.IntegratedSensor, 0, 0);
+        turretMotor.config_kP(0, TURRET_P);
+        turretMotor.config_kI(0, TURRET_I);
+        turretMotor.config_kD(0, TURRET_D);
+
+        hoodMotor.configForwardSoftLimitThreshold(MAX_HOOD_ANGLE * DEGREES_TO_ENCODER_TICKS);
+        hoodMotor.configReverseSoftLimitThreshold(0); // Current hood setup plan starts hood at 0, below MIN_HOOD_ANGLE
+        hoodMotor.configForwardSoftLimitEnable(true);
+        hoodMotor.configReverseSoftLimitEnable(true);
+        hoodMotor.configSupplyCurrentLimit(hoodCurrentLimit);
+        hoodMotor.setNeutralMode(NeutralMode.Brake);
+
+        this.flywheelMotor1 = flywheelMotor1;
+        this.flywheelMotor2 = flywheelMotor2;
+        this.turretMotor = turretMotor;
+        this.hoodMotor = hoodMotor;
+    }
+
+    /**
+     * Sets the target angle for the hood motor
+     * 
+     * @param degrees
+     *            Target angle for the hood motor in degrees
+     */
+    public void hoodMotorSetAngle(double degrees) {
+        degrees = Math.min(Math.max(degrees, MIN_HOOD_ANGLE), MAX_HOOD_ANGLE);
+        hoodMotor.set(ControlMode.Position, DEGREES_TO_ENCODER_TICKS * degrees);
+    }
+
+    /**
+     * Stops the hood motor
+     */
+    // TODO make hardstop
+    public void hoodMotorStop() {
+        hoodMotor.set(STOP_MOTOR);
+    }
+
+    /**
+     * Starts both flywheel motors
+     */
+    public void startFlywheel() {
+        flywheelMotor1.set(ControlMode.Velocity, FLYWHEEL_VELOCITY);
+    }
+
+    /**
+     * Stops both flywheel motors
+     */
+    public void stopFlywheel() {
+        flywheelMotor1.set(STOP_MOTOR);
+    }
+
+    /**
+     * Resets the turret motor's integrated encoder to 0.
+     */
+    // TODO use limit switches to reset the encoder
+    public void resetTurretEncoder() {
+        turretMotor.setSelectedSensorPosition(0);
+    }
+
+    /**
+     * Gets angle of the turret motor (horizontal swivel)
+     * 
+     * @return Angle, in degrees
+     */
+    public double getTurretAngle() {
+        return turretMotor.getSelectedSensorPosition() / DEGREES_TO_ENCODER_TICKS;
+    }
+
+    /**
+     * Sets the turret's angle to the given angle (Does not check angle limits).
+     * 
+     * @param angle
+     *            the angle (in degrees) to set the turret to (negative for
+     *            counterclockwise)
+     */
+    public void setTurretAngle(double angle) {
+        turretMotor.set(ControlMode.Position, DEGREES_TO_ENCODER_TICKS * angle);
```

Comment:
We may want to consider what behavior the turret has when it has rotated all the way to one side, but could still reach the target if it turned around to the other side.